### PR TITLE
Update egui with less wgpu lifetimes

### PR DESCRIPTION
**Pull request by emilk**
* Had this fix in it: https://github.com/emilk/egui/pull/5161
* https://github.com/rerun-io/rerun/issues/6801 is NOT fixed by this PR :(

@Wumpf you had a pretty solid repro of this yesterday - please test 🙏 

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7509?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7509?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!
* [x] If have noted any breaking changes to the log API in `CHANGELOG.md` and the migration guide

- [PR Build Summary](https://build.rerun.io/pr/7509)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: crates/viewer/re_renderer/src/queueable_draw_data.rs [hunk 1]
```diff
@@ -19,10 +19,10 @@ pub enum QueueableDrawDataError {
 
 type DrawFn = dyn for<'a, 'b> Fn(
         &Renderers,
-        &'b GpuRenderPipelinePoolAccessor<'b>,
+        &GpuRenderPipelinePoolAccessor<'a>,
```
Comment (emilk) [inline crates/viewer/re_renderer/src/queueable_draw_data.rs:22]: Could these lifetimes be given better names?

**Review by emilk — COMMENTED** [review]
Very nice wgpu cleanup!

**Comment by emilk** [comment]
@Wumpf says on slack that https://github.com/rerun-io/rerun/issues/6801 is still a problem:

image

and that happens both with `pixels_per_point=1` and `2`

File: crates/viewer/re_renderer/src/queueable_draw_data.rs (hunk 1 above)
Comment (Wumpf) [reply crates/viewer/re_renderer/src/queueable_draw_data.rs:22]: yes! That would be `'pipelines` and `'pass`

**Review by Wumpf — APPROVED** [review]
fixing up lifetime names, then good to merge regardless of its ability to fix the tooltip issue

**Pull request merged**